var/spack/repos/builtin/packages/mvapich2-gdr/package.py

```diff
@@ -161,28 +157,11 @@ def setup_run_environment(self, env):
 
         # Because MPI functions as a compiler, we need to treat it as one and
         # add its compiler paths to the run environment.
-        self.setup_compiler_environment(env)
+        self.setup_mpi_wrapper_variables(env)
 
     def setup_dependent_build_environment(self, env, dependent_spec):
-        self.setup_compiler_environment(env)
-        dependent_module = dependent_spec.package.module
-        env.set("MPICH_CC", dependent_module.spack_cc)
-        env.set("MPICH_CXX", dependent_module.spack_cxx)
-        env.set("MPICH_F77", dependent_module.spack_f77)
-        env.set("MPICH_F90", dependent_module.spack_fc)
-        env.set("MPICH_FC", dependent_module.spack_fc)
-
-    def setup_compiler_environment(self, env):
-        env.set("MPICC", join_path(self.prefix.bin, "mpicc"))
-        env.set("MPICXX", join_path(self.prefix.bin, "mpicxx"))
-        env.set("MPIF77", join_path(self.prefix.bin, "mpif77"))
-        env.set("MPIF90", join_path(self.prefix.bin, "mpif90"))
-
-    def setup_dependent_package(self, module, dependent_spec):
-        self.spec.mpicc = join_path(self.prefix.bin, "mpicc")
-        self.spec.mpicxx = join_path(self.prefix.bin, "mpicxx")
-        self.spec.mpifc = join_path(self.prefix.bin, "mpif90")
-        self.spec.mpif77 = join_path(self.prefix.bin, "mpif77")
+        self.setup_mpi_wrapper_variables(env)
+        super().setup_dependent_build_environment(env, dependent_spec)
```

Comment:
Potentially ambiguous use of `super` with multiple inheritance.

Reply:
Super is not ambiguous, it follows mro :wink:

Reply:
Sorry, ambiguous is the incorrect term here. I should have said Fragile as it is tied to the MRO, which changes if the order of parent classes changes.

Reply:
I don't have a preference between `super()` and an explicit call in this case. I'll change to call the base class method directly :wink:

Reply:
Missed this one too